### PR TITLE
Pn 26 pet owner login frontend

### PR DESCRIPTION
**Pull request by ShawnWang86b**

### CODE REVIEW

File: src/pages/RegisterPage.tsx [hunk 1]
```diff
@@ -1,7 +1,85 @@
...
+        <Text textColor="#4F4F4F" fontSize="28">
+          Already have a Pawshake account?
+        </Text>
+
+        <Text
+          textColor="#00AFED"
+          fontSize="16"
+          cursor="pointer"
+          _hover={{ textDecoration: "underline" }}
+          onClick={() => navigate("/login")}
```
Comment (jube33717f) [inline src/pages/RegisterPage.tsx:76]: same here

File: src/redux/api/apiSlice.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export const apiSlice = createApi({
+  baseQuery: fetchBaseQuery({ baseUrl: "http://localhost:5000/api" }),
```
Comment (jube33717f) [inline src/redux/api/apiSlice.ts:4]: baseUrl can be moved to a separate config file so that DevOps can set environment variable

**Review by jube33717f — COMMENTED** [review]
need to consider 'responsive'

**Review by jube33717f — COMMENTED** [review]
I personally feel that using inline styles can make the code harder to read and maintain. Provide some solutions below, but you can decide if you want to separate them or not.

1.Using the CSS prop to define styles in a separate file
`// MyComponent.js
import { Box } from '@chakra-ui/react';
import styles from './MyComponent.module.css';

function MyComponent() {
  return <Box CSS={styles.container}>Hello, world!</Box>;
}

// MyComponent.module.css
.container {
  background-color: #f5f5f5;
  border: 1px solid #ddd;
  padding: 10px;
  border-radius: 5px;
}
`
2. `withStyles`
`import { withStyle } from '@chakra-ui/react';

const MyBox = withStyle(Box, {
  baseStyle: {
    background-color: #f5f5f5;
   border: 1px solid #ddd;
   padding: 10px;
   border-radius: 5px;
  },
});

function MyComponent() {
  return <MyBox>Click me!</MyBox>;
}`
3. styled component way
`import { Button } from '@chakra-ui/react';
import { styled } from '@chakra-ui/system';

const CustomButton = styled(Button)({
  backgroundColor: '#4f4f4f',
  color: '#fff',
  padding: '8px 12px',
  borderRadius: '5px',
});

function MyComponent() {
  return <CustomButton>Click me!</CustomButton>;
}`

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "cSpell.words": ["Formik"]
+}
```
[the remark this answers is not in the document]
Comment (ShawnWang86b) [reply .vscode/settings.json]: I manually delete this file which includes "cSpell.words": ["Formik"].
I'm not adding the .vscode file into ignore, the reason is there have some settings, for example, auto format, auto-save

File: src/components/SignUp/SignUp.tsx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        try {
+          const { signUpUser }: any = await register({
+            firstName,
+            lastName,
+            userName,
+            email,
+            password,
+            phone,
+          }).unwrap();
+          console.log(signUpUser);
```
[the remark this answers is not in the document]
Comment (ShawnWang86b) [reply src/components/SignUp/SignUp.tsx]: Deleted

**Comment by ShawnWang86b** [comment]
after one day search and testing, I can't use styled-components with chakra properly, so I used variants instead in-line style.

**Comment by jube33717f** [comment]
> 
example here:
https://github.com/petNanny/pn-app/tree/styled-component-test/src/components/Login
in style:  `rem` is better than `px`
media queries:  `@media ${device.sm} ` is better than `@media (max-width: 768px) ` (https://www.mariokandut.com/how-to-use-media-queries-in-styled-components/)

**Comment by ShawnWang86b** [comment]
- add styled components for each of the components of login and sign up parts, followed by the example above
- add breakpoints, followed by the article above.
- refactor the folder structure, move test, styled files and component in each folder.

**Review by jube33717f — DISMISSED** [review]

**Review by jube33717f — APPROVED** [review]

**Pull request merged**